packages/expo-analytics-amplitude/android/src/main/java/expo/modules/analytics/amplitude/AmplitudeModule.java

```diff
@@ -90,6 +91,63 @@ public void setGroup(final String groupType, final List<Object> groupNames, Prom
     mClient.setGroup(groupType, new JSONArray(groupNames));
   }
 
+  @ExpoMethod
+  public void setTrackingOptions(final ReadableArguments options, Promise promise) {
+    if (rejectUnlessInitialized(promise)) {
```

Comment:
A comment on https://github.com/expo/expo/pull/5877/files#r334839075

Here, instead of rejecting unless initialized we could save the options and apply in between

```
mClient = new AmplitudeClient();
< here > 
mClient.initialize(getContext(), apiKey);
promise.resolve(null);
```
so that tracking options are applied before the client is initialized (with your implementation it is possible on iOS, just not on Android). Refactoring lines 100-146 to a separate function, remembering last set tracking options before initialization and applying options before setting API key should do the work. 🤔

Reply:
Good idea. https://github.com/expo/expo/pull/5877/commits/94d75cdbbb61f92f912025c5ce7b034b577f5cb8 should fix this